core/systems/threading/shared_thread.gd

```diff
@@ -101,6 +101,28 @@ func scheduled_exec(method: Callable, wait_time_ms: int) -> void:
 	mutex.unlock()
 
 
+## Cancels a given Sheduled Task before it is executed.
+func cancel_scheduled_exec(task: ScheduledTask) -> void:
+	mutex.lock()
+	if task not in scheduled_funcs:
```

Comment:
You should lock the mutex, duplicate, then check if the task exists to minimize the amount of time the mutex is locked, which will block the thread processing.

Reply:
I'll just need to lock it again right after?

Reply:
Unlock it, yeah. Lock, duplicate, then unlock. Otherwise the array could get mutated from another thread while it is being read.